packages/auth-remix/src/server.ts

```diff
@@ -0,0 +1,795 @@
+        switch (path) {
+          case "oauth": {
+            if (!onOAuthCallback) {
+              throw new Error(
+                `'onOAuthCallback' auth route handler not configured`
+              );
+            }
+            const provider = searchParams.get(
+              "provider_name"
+            ) as BuiltinOAuthProviderNames | null;
+
+            if (!provider || !builtinOAuthProviderNames.includes(provider)) {
+              throw new Error(`invalid provider_name: ${provider}`);
+            }
+            const redirectUrl = `${this._authRoute}/oauth/callback`;
+            const pkceSession = await this.core.then((core) =>
+              core.createPKCESession()
+            );
+
+            return redirect(
+              pkceSession.getOAuthUrl(
+                provider,
+                redirectUrl,
+                `${redirectUrl}?isSignUp=true`
+              ),
+              {
+                headers: new Headers({
+                  "Set-Cookie": `${this.options.pkceVerifierCookieName}=${pkceSession.verifier}; HttpOnly;`,
+                }),
+              }
+            );
+          }
+          case "oauth/callback": {
+            if (!onOAuthCallback) {
+              throw new Error(
+                `'onOAuthCallback' auth route handler not configured`
+              );
+            }
+            const error = searchParams.get("error");
+            if (error) {
+              const desc = searchParams.get("error_description");
+              return onOAuthCallback({
+                error: new Error(error + (desc ? `: ${desc}` : "")),
+              });
+            }
+            const code = searchParams.get("code");
+            const isSignUp = searchParams.get("isSignUp") === "true";
+
+            const verifier =
+              parseCookies(request)[this.options.pkceVerifierCookieName];
+
+            if (!code) {
+              return onOAuthCallback({
+                error: new Error("no pkce code in response"),
+              });
+            }
+
+            if (!verifier) {
+              return onOAuthCallback({
+                error: new Error("no pkce verifier cookie found"),
+              });
+            }
+            let tokenData: TokenData;
+            try {
+              tokenData = await (await this.core).getToken(code, verifier);
+            } catch (err) {
+              return onOAuthCallback({
+                error: err instanceof Error ? err : new Error(String(err)),
+              });
+            }
+
+            const headers = new Headers();
+            headers.append(
+              "Set-Cookie",
+              `${this.options.authCookieName}=${tokenData.auth_token}; HttpOnly; SameSite=Lax; Path=/`
+            );
+            headers.append(
+              "Set-Cookie",
+              `${this.options.pkceVerifierCookieName}=`
+            );
+
+            return oathCallbackCall(onOAuthCallback, headers, {
```

Comment:
Maybe this function should have a different name? since it's not just used for oauth.

Reply:
yes, do u think its better to use authRouteHandlerCbCall, and for the other one to use cbCall instead of callbackCall?